src/java/org/apache/cassandra/service/accord/AccordCommandStore.java

```diff
@@ -505,32 +507,27 @@ else if (!CommandsForKey.managesExecution(txnId) && command.hasBeen(PreCommitted
         }
 
         @Override
-        public void load(Command command, OnDone onDone)
+        public AsyncChain<Void> load(TxnId txnId, Supplier<Command> supplier)
         {
-            store.execute(context(command, SYNC), safeStore -> loadInternal(command, safeStore))
-                 .begin((unused, throwable) -> {
-                     if (throwable != null)
-                         onDone.failure(throwable);
-                     else
-                         onDone.success();
-                 });
+            if (store.caches.commands().isReferenced(txnId))
+                return AsyncResults.SUCCESS_NULL;
+
+            Command command = supplier.get();
+            return store.submit(context(command, SYNC),
```

Comment:
Is the `commandStore` going to load this also by reading from the journal because the command is in its context? Might it be easier even to just leave it entirely to the `commandStore`?